src/server/saved_objects/migrations/core/elastic_index.ts

```diff
@@ -284,3 +282,32 @@ async function assertIsSupportedIndex(indexInfo: FullIndexInfo) {
   }
   return indexInfo;
 }
+
+/**
+ * Reindexes from source to dest, polling for the reindex completion.
+ */
+async function reindex(callCluster: CallCluster, source: string, dest: string, batchSize: number) {
+  // We poll instead of having the request wait for completion, as for large indices,
+  // the request times out on the Elasticsearch side of things. We have a relatively tight
+  // polling interval, as the request is fairly efficent, and we don't
+  // want to block index migrations for too long on this.
+  const pollInterval = 250;
+  const { task } = await callCluster('reindex', {
+    body: {
+      dest: { index: dest },
+      source: { index: source, size: batchSize },
+    },
+    refresh: true,
+    waitForCompletion: false,
+  });
+
+  let completed = false;
+
+  while (!completed) {
+    await new Promise(r => setTimeout(r, pollInterval));
+
+    completed = await callCluster('tasks.get', {
+      taskId: task,
+    }).then(result => result.completed);
```

Comment:
thoughts on adding a catch to cancel the loop if there's an error?

Reply:
It will exit the loop via the exception (we're doing an await), so I think we're safe as is.